File: frontend/src/types/Note.ts
import type { Tag } from './Tag'

export interface Note {
  id: number
  title: string
  content: string
  isPinned: boolean
  isArchived: boolean
  tags: Tag[]
  createdAt: string
  updatedAt: string
}

export interface NoteQuery {
  keyword?: string
  tagId?: number
  isPinned?: boolean
  isArchived?: boolean
  page?: number
  size?: number
}

export interface NotePayload {
  title: string
  content: string
  tagIds?: number[]
}

File: backend/src/main/java/com/aireview/entity/Note.java
package com.aireview.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 笔记实体。
 */
@Data
@TableName("note")
public class Note {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String title;

    private String content;

    private Boolean isPinned;

    private Boolean isArchived;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
